cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -126,15 +172,238 @@ fn check_builder_metadata(_builder: impl ManifestBuilder) -> Result<CheckedBuild
     todo!()
 }
 
+fn gather_build_repo_meta(environment: &PathEnvironment) -> Result<LockedUrlInfo, PublishError> {
+    // Gather build repo info
+    let git = match environment.parent_path() {
+        Ok(env_path) => match GitCommandProvider::discover(env_path) {
+            Ok(git) => git,
+            Err(e) => return Err(PublishError::UnsupportEnvironmentState { msg: e.to_string() }),
+        },
+        Err(e) => return Err(PublishError::UnsupportEnvironmentState { msg: e.to_string() }),
+    };
+
+    let origin = match git.get_origin() {
+        Ok(origin) => origin,
+        Err(e) => return Err(PublishError::UnsupportEnvironmentState { msg: e.to_string() }),
+    };
+    let rev = match origin.revision {
+        Some(rev) => rev,
+        None => {
+            return Err(PublishError::UnsupportEnvironmentState {
+                msg: "No revision found".to_string(),
```

Comment:
These are kind of just equivalent ways to do the same thing you do here with a `match` statement.
I think unless there is more involved handling of either case we've been using these more frequently than the full match statement.

suggestion (style): using `ok_or` shorthand


```suggestion  
  let rev = origin
        .revision
        .ok_or(PublishError::UnsupportEnvironmentState {
            msg: "No revision found".to_string(),
        })?;
```

suggestion(style): using let _ else

```suggestion
    let Some(rev) = origin.revision else {
        return Err(PublishError::UnsupportEnvironmentState {
            msg: "No revision found".to_string(),
        });
    }
```